x/ecocredit/params.go

```diff
@@ -37,12 +45,45 @@ func validateCreditClassFee(i interface{}) error {
 	return nil
 }
 
-func NewParams(creditClassFee sdk.Coins) Params {
+func validateCreditTypes(i interface{}) error {
+	v, ok := i.([]*CreditType)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	// ensure no duplicate credit types and that all precisions conform to hardcoded PRECISION above
+	seenTypes := make(map[string]bool)
+	for _, ct := range v {
+		t := strings.ToLower(ct.Type)
+		t = util.FastRemoveWhitespace(t)
+
+		// TODO: remove after we open governance changes for precision
+		if ct.Precision != PRECISION {
+			return fmt.Errorf("invalid precision %d: precision is currently lockd to %d", ct.Precision, PRECISION)
```

Comment:
I'm not sure what context this is called in, but is there an `sdkerrors` error type that would be appropriate to `Wrap` here?

Reply:
probably invalidrequest - will only really be called during a tx lifecycle for params